components/script/dom/xmlhttprequest.rs

```diff
@@ -1450,3 +1478,14 @@ pub fn is_field_value(slice: &[u8]) -> bool {
         }
     })
 }
+
+// Serialize a Document struct
+fn serialize_document(document: &Document) -> Fallible<DOMString> {
+    let mut writer = vec![];
+    if let Ok(()) = serialize(&mut writer, &document.upcast::<Node>(),
+        SerializeOpts { traversal_scope: ChildrenOnly, ..Default::default() }) {
```

Comment:
IIRC `traversal_scope` is `ChildrenOnly` by default, so the second argument can just be `Default::default()` instead.